### ISSUE TITLE
Feature suggest: Add an "is-probing" pin to motmod?

### ISSUE THREAD

**Issue by ReeseWang** [LinuxCNC/linuxcnc#371]
I'm writing some auto measurement G-code for my milling machine with a touch probe.
I definitely don't want to crash the probe.
But LinuxCNC (2.7.11) seems don't want to stop when the probe is tripped during a non-probe move if the machine is running G-code program (AUTO mode IIRC).

**Comment by SebKuzminsky** [comment]
I think the goal of having accidental/unexpected probe trips abort any motion is a good idea.

Does @dngarrett 's recent fix in master fix this for you?  It's commit e10c52ea, available in v2.8.0-pre1-3405-ge10c52e and later.

If not, can you add a note to this issue, or to #368 and/or #369?

Also note that motion has a pin named motion.motion-type, which is set to 5 only when in a probing move: http://linuxcnc.org/docs/devel/html/man/man9/motion.9.html#MOTION PINS

**Comment by rene-dev** [comment]
the other issues only address stopping during homing/jogging/mdi, but not auto mode.
I just checked, and it does not care about the probe during auto mode.
stopping during any non-probe move anytime sounds good.

**Comment by Lcvette** [comment]
I agree, i am on version 2.9 but also tried with 2.8, running debian 9 the stretch test iso, i have written a entire probing routine library for the new probe_basic gui but have been getting reports along with my own testing of the probe trigger during non probe moves not caring and causing crashes. I am assuming nothing has been fixed yet?  if so is there a way to fix it or a work around?  maybe something added to the ngc probe routine file?

thanks in advance!

Chris

**Comment by rmu75** [comment]
Is this still relevant? I'm pretty sure I recently had a stop caused by probe contact while jogging.

**Comment by andypugh** [comment]
This was about probe-trip in auto mode. 

Using a sim config of 2.9 and using halshow to toggle the probe pin I have determined that toggling the probe halts motion in continuous jogging, incremental jogging, running g-code and MDI. 

There may still be an argument for a dedicated pin that can be used to "gate" the jogging input. Some systems have table-mounted tool height measurement devices and these can easily be triggered by vibration. (as comparing the motion-type pin is a bit of an untidy bit of HAL involving type conversions and comparators.

**Comment by rmu75** [comment]
I would just "and" that probe input either in hal, classicladder or externally in hardware with a digital output you can set in the probing routine. It could also make sense to add a separate probe input pin that doesn't stop the machine when triggered in a non-probe move. Then you can decide if you prefer crashing your probe or ruining your workpeace by spurious aborts ;-)

**Comment by andypugh** [comment]
There are existing INI config settings to disable the abort in some cases:
NO_PROBE_JOG_ERROR = 0 - Allow to bypass probe tripped check when you jog manually.
NO_PROBE_HOME_ERROR = 0 - Allow to bypass probe tripped check when homing is in progress.
